content/4.examples/query-payment-states.md

```diff
@@ -0,0 +1,138 @@
+# Query payment states
+
+Corda stores the state of a payment in the vault of the node that initiated the payment. So you can access the state of a payment by querying the vault.
+
+```mermaid
+classDiagram
+  class MyPaymentService {
+    + queryPaymentsViaGevamu()
+  }
+  class GevamuFacade {
+    + getPayment()
+    + queryPayments()
+  }
+
+  MyPaymentService ..> GevamuFacade
+```
+
+## Payment structure
+
+Structure of the [`Payment`](https://gevamu.github.io/corda-payments-sdk/payments-contracts/com.gevamu.corda.states/-payment/index.html) is as follows:
+
+::notice{info}
+  Pay attention, that full payment instruction is stored as Corda [`Attachment`](https://docs.r3.com/en/api-ref/corda/4.8/open-source/javadoc/net/corda/core/contracts/Attachment.html). So you need to download it and deserialize it to get all payment details.
+::
+
+```mermaid
+classDiagram
+  class Payment {
+    + payer: Party
+    + gateway: Party
+    + endToEndId: String
+    + paymentInstructionId: AttachmentId
+    + status: PaymentStatus
+    + additionalInfo: String?
+    + uniquePaymentId: UUID
+    + timestamp: Instant
+  }
+
+  class PaymentStatus{
+    <<enumeration>>
+    CREATED
+    SENT_TO_GATEWAY
+    ACCEPTED
+    PENDING
+    COMPLETED
+    REJECTED
+  }
+
+  class Attachment {
+    <<interface>>
+    + id: SecureHash
+    + open(): InputStream
+  }
+
+  Payment o-- PaymentStatus
+  Payment .. Attachment: paymentInstructionId --> id
+```
+
+## About vault queries
+
+Every Corda node have a vault that stores all states that are created by the node. Our interest is in the `Payment` states.
+
+Corda provides a powerful query language to search for states in the vault. You can find more information about it in the [Writing vault queries](https://docs.r3.com/en/platform/corda/4.7/enterprise/cordapps/api-vault-query.html) section of the documentation.
+
+::notice{warning}
+  Payment attachments are stored as files. So you can't use vault queries to search by payment details stored in attachments.
```

Comment:
```suggestion
However, you cannot use vault queries to search by payment details inside Payment attachments, as they are stored as files.
```

Reply:
Gevamu SDK doesn't provide ready to use solution for querying by payments details stored in the attachment. However, you can create additional tables with the information needed for your business needs and use them in vault queries.

@lingvolab please proofread it

Reply:
```suggestion
  Gevamu SDK doesn't provide any ready-to-use solution for querying by payments details stored in the attachment. However, you can create additional tables with the information needed for your business needs and use them in vault queries.
```

Reply:
```suggestion
  Gevamu SDK doesn't provide any ready-to-use solution for querying by payments details stored in the attachment. However, you can create additional tables with the information needed for your business needs and use them in vault queries. 
  To get all payment details from the attachment, download and deserialize it.
 
```